### ISSUE TITLE
/transmit/ route complains when transmitting a single Info

### ISSUE THREAD

**Issue by ddbourgin** [Dallinger/Dallinger#278]
I am attempting to transmit a single `Info` to the psiturk server using Dallinger's `/info/ + <node_id> + /transmit` route. Unfortunately, when I try to do this using the code snippet below, the request does not go through. 
## Current Behavior

I am attempting to transmit a single `Info` using Dallinger's `/info/ + <node_id> + /transmit` route. I am using the following `POST` requests in my `experiment.js` file:

``` javascript
submit_response = function() {
    info = {'this': ['is'], 'a': ['test']};
    my_node_id = sessionStorage.my_node_id;

    reqwest({
        url: "/info/" + my_node_id,
        method: 'post',
        data: {
            contents: JSON.stringify(info),
            info_type: "Info"
        },
        success: function (resp) {
            transmit_info();
        },
        error: function (err) {
            console.log(err);
            err_response = JSON.parse(err.response);
            $('body').html(err_response.html);
        }
    });
}

transmit_info = function() {
    my_node_id = sessionStorage.my_node_id;
    reqwest({
        url: "/node/" + my_node_id + "/transmit",
        method: 'post',
        success: function (resp) {
            console.log(resp);
        },
        error: function (err) {
            console.log(err);
            err_response = JSON.parse(err.response);
            $('body').html(err_response.html);
        }
    });
}
```

Upon calling the `transmit_info` function, the psiturk server log shows the following traceback:

``` python
2016-10-25 13:22:30 [58329] [DEBUG] POST /node/2/transmit
2016-10-25 13:22:30,613 POST /node/2/transmit
Traceback (most recent call last):
  File "/private/tmp/tmpwDlHOG/3f849e7c-16e5-4d45-922a-2cc134922925/custom.py", line 977, in node_transmit
    for t in transmissions:
TypeError: 'Transmission' object is not iterable
```

On further investigation, it appears that line 1184 in `dallinger/models.py` may be the culprit, wherein the `node.transmit` method is explicitly converting transmissions to a single `Transmission` instance rather than a list:

``` python
if len(transmissions) == 1:
    return transmissions[0]
else:
    return transmissions
```

My question: What is the purpose of this behavior ? More generally, is there a better way for me to be performing this request?

As a workaround, I have added the following line to `custom.py` to ensure that `transmissions` is always iterable:

``` python
transmissions = node.transmit(what=what, to_whom=to_whom)

if not isinstance(transmissions, list):
    transmissions = [transmissions]

for t in transmissions:
    assign_properties(t)
```

This appears to work fine in my case, although may be suboptimal in general.
## Local environment:

``` bash
> uname -a
Darwin ddb-mbp.local 14.5.0 Darwin Kernel Version 14.5.0: Thu Jun 16 19:58:21 PDT 2016; root:xnu-2782.50.4~1/RELEASE_X86_64 x86_64
```

``` bash
> dallinger --version
2.3.1
```

``` bash
> pip list
alabaster (0.7.9)
appnope (0.1.0)
APScheduler (3.2.0)
Babel (2.3.4)
backports.shutil-get-terminal-size (1.0.0)
boto (2.15.0)
click (6.6)
cmd2 (0.6.7)
codecov (2.0.5)
coverage (4.2)
dallinger (2.3.1)
decorator (4.0.10)
docopt (0.6.1)
docutils (0.12)
enum34 (1.1.6)
fake-factory (0.3.2)
Flask (0.10.1)
funcsigs (1.0.2)
future (0.15.2)
futures (3.0.5)
fuzzywuzzy (0.2)
GitPython (0.1.7)
gunicorn (18.0)
imagesize (0.7.1)
ipython (5.1.0)
ipython-genutils (0.1.0)
itsdangerous (0.24)
Jinja2 (2.8)
localconfig (0.4.2)
MarkupSafe (0.23)
nose (1.3.7)
numpy (1.11.2)
pathlib2 (2.1.0)
pexpect (4.2.1)
pickleshare (0.7.4)
pip (8.1.2)
prompt-toolkit (1.0.7)
psiturk-dallinger (3.1.3)
psutil (1.2.1)
psycopg2 (2.6.2)
ptyprocess (0.5.1)
pyenchant (1.6.8)
Pygments (2.1.3)
pypandoc (1.2.0)
pyparsing (2.1.10)
pytz (2016.7)
redis (2.10.5)
requests (2.2.1)
rq (0.6.0)
setproctitle (1.1.8)
setuptools (28.3.0)
sh (1.9)
simplegeneric (0.8.1)
six (1.10.0)
snowballstemmer (1.2.1)
Sphinx (1.4.8)
sphinxcontrib-spelling (2.2.0)
SQLAlchemy (1.1.0)
traitlets (4.3.1)
tzlocal (1.2.2)
ua-parser (0.3.6)
user-agents (0.2.0)
wcwidth (0.1.7)
Werkzeug (0.11.11)
wheel (0.29.0)
```

**Comment by thomasmorgan** [comment]
David is right, this is unhelpful behavior.

I suggest replacing
```
if len(transmissions) == 1:
            return transmissions[0]
        else:
            return transmissions
```
with just
```return transmissions
```
this makes it a bit ugly as if you ask a node to transmit a single info to a single node it will return a list containing a single transmission. But the big problem this would solve is that if you ask a node to transmit but allow default behavior (_what and _to_whom) you can't actually predict whether you will get a single transmission or a list of transmissions, better to always use a list.

**Comment by thomasmorgan** [comment]
@ddbourgin your code and your fix is reasonable. I think you have encountered this error as no current experiments have used the node/<id>/transmit route before and so its somewhat untested.

Another solution is to overwrite the info_post_request method in experiments.py. Basically this function runs whenever someone makes an info post request and you could put the node.transmit() call in there, this would avoid the buggy transmit route. However, bear in mind this runs every time someone posts and info so if you only occasionally want to transmit this might require some conditions attached to it.

**Comment by ddbourgin** [comment]
Awesome! Thanks for this. I'm happy to submit a PR with the change to `node.transmit` if that's useful. Otherwise, I'll just use my hacked version :)

**Comment by thomasmorgan** [comment]
Use the hack for now. I'll discuss this with @suchow before creating a fix.

**Comment by suchow** [comment]
Closed by #656.
